builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
That seems correct. The final image you can just handle after dispatchers have completed.